config/src/main/java/org/springframework/security/config/annotation/web/configurers/AnonymousConfigurer.java

```diff
@@ -142,6 +148,19 @@ public AnonymousConfigurer<H> authenticationFilter(AnonymousAuthenticationFilter
 		return this;
 	}
 
+	/**
+	 * Specifies a custom {@link AuthenticationDetailsSource} to use for anonymous
+	 * authentication.
+	 * @param authenticationDetailsSource the custom {@link AuthenticationDetailsSource}
+	 * to use
+	 * @return {@link AnonymousConfigurer} for additional customization
+	 */
+	public AnonymousConfigurer<H> authenticationDetailsSource(
```

Comment:
Instead of adding another DSL configuration, let's consider adding post-processing support for `AnonymousAuthenticationFilter`. In this way, folks will be able to configure this and other properties as well.

Since the DSL already allows for the complete filter to be provided, we want to take care in exposing each individual component, especially those that are much less-frequently configured.